### PR TITLE
Implement bit shift operations in BitSet

### PR DESCRIPTION
**Pull request by Heraldk**
### Checklist

- [x] I've looked at the [contribution guidelines](https://github.com/raywenderlich/swift-algorithm-club/blob/master/.github/CONTRIBUTING.md).
- [x] This pull request is complete and ready for review.

### CODE REVIEW

File: Bit Set/BitSet.swift [hunk 1]
```diff
@@ -221,6 +221,51 @@ prefix public func ~ (rhs: BitSet) -> BitSet {
...
 
+// MARK: - Bit shift operations
+
+/*
+ Note: For bitshift operations, the assumption is that any bits that are
+ shifted off the end of the end of the declared size are not still set.
+ In other words, we are maintaining the original number of bits.
+ */
+
+public func << (lhs: BitSet, numBitsLeft: Int) -> BitSet {
```
Comment (richard-ash) [inline Bit Set/BitSet.swift]: Looks like you already implemented this method above. Is there a reason for the duplication?

File: Bit Set/BitSet.swift [hunk 2]
```diff
@@ -221,6 +221,51 @@ prefix public func ~ (rhs: BitSet) -> BitSet {
...
+        if (i - offset - 1 >= 0) {
+            out.words[i] |= lhs.words[i - offset - 1] >> (lhs.N - shift)
+        }
+    }
+    
+    out.clearUnusedBits()
+    return out
+}
+
+public func >> (lhs: BitSet, numBitsRight: Int) -> BitSet {
```
Comment (richard-ash) [inline Bit Set/BitSet.swift]: Same here

File: Bit Set/BitSet.swift (hunk 1 above)
Comment (Heraldk) [reply Bit Set/BitSet.swift]: For both of these, I'm updating the source that's in the playground as well as the main source file. I couldn't find an obvious way these two files were linked (if they are at all) so it seemed like I needed to update both source files to match.
Unless I'm missing something about how playgrounds work?

**Comment by Heraldk** [comment]
> Code looks good 👍 Left one minor comment inline. Would you mind adding a section in the README explaining these two methods?
> 
> Thanks again!

Sure, I'll add some README notes soon. Might be a day or two :)

File: Bit Set/BitSet.swift (hunk 1 above)
Comment (richard-ash) [reply Bit Set/BitSet.swift]: Hmm yea, that's a good point. Would you mind deleting the entire file? (`Bit Set/BitSet.swift`) Ideally, we'd want only a single set of files with all the source code

**Comment by richard-ash** [comment]
> > Code looks good 👍 Left one minor comment inline. Would you mind adding a section in the README explaining these two methods?
> > Thanks again!
> 
> Sure, I'll add some README notes soon. Might be a day or two :)

Sounds good! Looking forward to reading it 🙂

File: Bit Set/BitSet.swift (hunk 1 above)
Comment (Heraldk) [reply Bit Set/BitSet.swift]: Is that the new pattern? There are a lot of instances in this repository where the code is duplicated in the playground either as an inclusion or just the code copy-pasted into the playground source file.

I can remove the file outside the playground if you like :)

**Comment by Heraldk** [comment]
I added a README section for bit-shift operations, and removed the duplicate source file as requested.

File: Bit Set/BitSet.swift (hunk 1 above)
Comment (richard-ash) [reply Bit Set/BitSet.swift]: Yea was looking around an noticed this too. I'll start pointing it out more as people update the repo. Even if the code is copy-pasted its a super easy thing to forget to update. Much easier to just have one source of truth.

That'd be great thanks :)

**Comment by richard-ash** [comment]
> I added a README section for bit-shift operations, and removed the duplicate source file as requested.

Thanks, 😄the section looks good, nice work!

**Comment by Heraldk** [comment]
Anything left on me that prevents this from being merged? No rush - just making sure I'm not missing something :)

**Comment by richard-ash** [comment]
Oops thanks for the reminder

**Pull request merged**